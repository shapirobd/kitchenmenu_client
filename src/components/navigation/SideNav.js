import React from "react";
import { useStyles } from "./styles/SideNavStyles";
import { Button } from "@material-ui/core";
import ArrowRightIcon from "@material-ui/icons/ArrowRight";
import { useSelector } from "react-redux";
import SideNavDrawer from "./SideNavDrawer";

const SideNav = () => {
	const classes = useStyles();
	const [state, setState] = React.useState({
		left: false,
	});

	const toggleDrawer = (anchor, open) => (event) => {
		if (
			event &&
			event.type === "keydown" &&
			(event.key === "Tab" || event.key === "Shift")
		) {
			return;
		}

		setState({ ...state, [anchor]: open });
	};

	const user = useSelector((state) => state.user);

	const btnProps = {
		name: "sideBtn",
	};

	return user ? (
		<React.Fragment>
			<Button onClick={toggleDrawer("left", true)} className={classes.toggler}>
				<ArrowRightIcon />
			</Button>
			<SideNavDrawer state={state} toggleDrawer={toggleDrawer} user={user} />
		</React.Fragment>
	) : null;
};

export default SideNav;
