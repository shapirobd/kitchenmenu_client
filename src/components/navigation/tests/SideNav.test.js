import React from "react";
import { render, screen } from "@testing-library/react";
import SideNav from "../SideNav";
import { MemoryRouter } from "react-router-dom";
import { PersistGate } from "redux-persist/integration/react";
import { Provider } from "react-redux";
import { store, persistor } from "../../../configureStore";

it("should render without crashing", () => {
	render(
		<Provider store={store}>
			<MemoryRouter>
				<PersistGate loading={null} persistor={persistor}>
					<SideNav />
				</PersistGate>
			</MemoryRouter>
		</Provider>
	);
});

it("should match screenshot", () => {
	const { asFragment } = render(
		<Provider store={store}>
			<MemoryRouter>
				<PersistGate loading={null} persistor={persistor}>
					<SideNav />
				</PersistGate>
			</MemoryRouter>
		</Provider>
	);

	expect(asFragment()).toMatchSnapshot();
});
