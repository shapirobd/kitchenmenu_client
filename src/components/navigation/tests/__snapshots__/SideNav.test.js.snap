// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match screenshot 1`] = `<DocumentFragment />`;
